homeassistant/helpers/script.py

```diff
@@ -394,19 +395,19 @@ async def async_run(self) -> None:
             self._finish()
 
     async def _async_step(self, log_exceptions):
+        continue_on_error = self._action.get(CONF_CONTINUE_ON_ERROR, False)
```

Comment:
Is there any reason why we dont default to `true`? I think reasonably a new user would expect actions to continue if one were to fail. I know that we had a PR to stop on error and now we want to allow continuing but in all honesty if we had an automation that was turning on lights/switches or sending a notification to a few different providers it doesnt make sense to stop at the first failure. Reasonably if a user adds more than 1 step they would expect it all to continue unless they used a condition or something similar.  Errors are not something a user should anticipate when creating an automation so to them an error would be an edge case or something that doesn't happen often.

Reply:
> Is there any reason why we dont default to true? 

Yes :)

> I think reasonably a new user would expect actions to continue if one were to fail. 

I don't agree. Silencing errors are problematic. It causes searching and frustration on why something doesn't work as expected.

Therefore, we have been moving towards raising more often in the past years. For example, try using a non-existing variable in a template. Or calling a non-existing service. And many more of those.

If anything, I wish we could make it more clear the last run errored/failed (e.g., showing a column on the automations/script dashboard that shows that).